gensim/models/keyedvectors.py

```diff
@@ -323,34 +323,16 @@ def get_vecattr(self, key, attr):
         index = self.get_index(key)
         return self.expandos[attr][index]
 
-    def resize_vectors(self):
-        """Make underlying vectors match index_to_key size."""
-        target_count = len(self.index_to_key)
-        prev_count = len(self.vectors)
-        if prev_count == target_count:
-            return ()
-        prev_vectors = self.vectors
-        if hasattr(self, 'mapfile_path') and self.mapfile_path:
-            self.vectors = np.memmap(self.mapfile_path, shape=(target_count, self.vector_size), mode='w+', dtype=REAL)
-        else:
-            self.vectors = np.zeros((target_count, self.vector_size), dtype=REAL)
-        self.vectors[0: min(prev_count, target_count), ] = prev_vectors[0: min(prev_count, target_count), ]
-        self.allocate_vecattrs()
-        self.norms = None
-        return range(prev_count, target_count)
+    def resize_vectors(self, seed=0):
+        """Make underlying vectors match index_to_key size; random-initialize any new rows."""
 
-    def randomly_initialize_vectors(self, indexes=None, seed=0):
-        """Initialize vectors with low-magnitude random vectors, as is typical for pre-trained
-        Word2Vec and related models.
+        target_shape = (len(self.index_to_key), self.vector_size)
+        self.vectors = prep_vectors(target_shape, prior_vectors=self.vectors, seed=seed)
+        # TODO: support memmap?
```

Comment:
In my opinion, if this is a future idea, we should add it as a ticket. If this is dead code, we should remove it. Adding TODOs like this decreases the quality of the code.

Reply:
On this particular case, yes - there's already a pending issue (#2955) for making mmap stuff work as well or better than it used to, before 4.0.0, and these are some of the exact lines where the code is almost certain to change/expand, so this is just temporary ugliness to draw attention here until that's done. 

More generally: I think there are many, many smaller potential improvements or "maybe this could/should change" observations that are much better highlighted with code comments, including intentions like "TODO MAYBE" or "FIXME?", than issues. They may not deserve any current attention, or limited triage bandwidth, via appearing in lists-of-issues. They may only make sense in the context of exact micro-decisions in the code that are a pain to link/restate in issue prose elsewhere. (And thus, they'll be lost if the standard is, "go to Github in a web browser & make a proper issue".) But, they become intensely relevant if and only if someone else (or the original author) is actually reviewing the same lines-of-code.